src/rules.jl

```diff
@@ -248,11 +248,15 @@ _abs_deriv(x) = signbit(x) ? -one(x) : one(x)
 @define_diffrule LogExpFunctions.logmxp1(x) = :((1 - $x) / $x)
 
 # binary
-@define_diffrule LogExpFunctions.xlogy(x, y) = :(log($y)), :($x / $y)
+@define_diffrule LogExpFunctions.xlogy(x, y) = 
+    :(log($y)),
+    :(z = $x / $y; iszero($x) && !isnan($y) ? zero(z) : iszero($y) ? oftype(z, NaN) : z)
```

Comment:
Maybe we rather want

```suggestion
    :(z = $x / $y; iszero($x) && !isnan($y) ? zero(z) : z)
```

here. See https://github.com/JuliaStats/LogExpFunctions.jl/pull/54#discussion_r916492896. An additional reason is that otherwise `z = NaN` might be changed to a different `NaN`.

Reply:
This is what I had before. The NaN conventions are misterious to me. I think the -Inf is more informative, but this will make it impossible to match with finite differencing (I think).